### PR TITLE
New Resource azurerm_healthbot

### PR DESCRIPTION
**Pull request by haiyuazhang**
Support for [Azure health bot Service](https://docs.microsoft.com/en-us/azure/health-bot/)

The ACC Test PASS log:
```
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/bot -run=TestAccBotHealthbot* -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/04/21 14:03:44 [DEBUG] not using binary driver name, it's no longer needed
2021/04/21 14:03:46 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccBotHealthbot_basic
=== PAUSE TestAccBotHealthbot_basic
=== RUN   TestAccBotHealthbot_requiresImport
=== PAUSE TestAccBotHealthbot_requiresImport
=== RUN   TestAccBotHealthbot_complete
=== PAUSE TestAccBotHealthbot_complete
=== RUN   TestAccBotHealthbot_update
=== PAUSE TestAccBotHealthbot_update
=== CONT  TestAccBotHealthbot_basic
=== CONT  TestAccBotHealthbot_update
=== CONT  TestAccBotHealthbot_complete
=== CONT  TestAccBotHealthbot_requiresImport
--- PASS: TestAccBotHealthbot_complete (202.30s)
--- PASS: TestAccBotHealthbot_basic (205.21s)
--- PASS: TestAccBotHealthbot_requiresImport (216.32s)
--- PASS: TestAccBotHealthbot_update (331.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/bot 332.966s
```

### CODE REVIEW

File: .teamcity/components/generated/services.kt [hunk 1]
```diff
@@ -38,6 +38,7 @@ var services = mapOf(
         "hdinsight" to "HDInsight",
         "hpccache" to "HPC Cache",
         "hsm" to "Hardware Security Module",
+        "healthbot" to "Health Bot",
```
Comment (katbyte) [inline .teamcity/components/generated/services.kt:41]: Should this be part of the healthcase or bot service?

Comment (katbyte) [reply .teamcity/components/generated/services.kt:41]: or will the be a bunch of new resources here?

File: azurerm/internal/services/healthbot/healthbot_service_resource.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	id := parse.NewHealthbotID(subscriptionId, resourceGroup, name).ID()
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
```
Comment (katbyte) [inline azurerm/internal/services/healthbot/healthbot_service_resource.go:85]: could we use the id where we can?
```suggestion
		existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
```

File: azurerm/internal/services/healthbot/healthbot_service_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccHealthbot_complete(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/healthbot/healthbot_service_resource_test.go:47]: the complete test shjould only have 1 step - the update test should go basiuc -> complete -> basic

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @haiyuazhang 0 overall this looks good btu i've left some questions inline to address

File: .teamcity/components/generated/services.kt (hunk 1 above)
Comment (haiyuazhang) [reply .teamcity/components/generated/services.kt:41]: The swagger spec sits [here](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/healthbot/resource-manager
). So I guess It's not part of healthcare or bot service.

Only one resource is defined in the spec.

Comment (katbyte) [reply .teamcity/components/generated/services.kt:41]: given its basically bot/chat ai like the bot service it would make sense to put it under bots?

Comment (haiyuazhang) [reply .teamcity/components/generated/services.kt]: I've made changes according to your comments please help to review the the changes. thanks.

**Review by katbyte — CHANGES_REQUESTED** [review]
thanks @haiyuazhang - i think `azurerm_bot_healthbot` is a bit redundant as it has bot in it twice, if we change this to `azurerm_healthbot` i think this will be good to merge!

**Comment by haiyuazhang** [comment]
@katbyte 
I've rename the resource to azurerm_healthbot. Besides, I've rebased the code from master branch to fix the deps check error caused by the sdk version upgrade in master branch. Thanks for your comments and review.

File: website/docs/r/bot_healthbot.html.markdown [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported:
+
+* `id` - The ID of the resource.
+
+* `bot_management_portal_link` - The management link.
```
Comment (katbyte) [inline website/docs/r/bot_healthbot.html.markdown:57]: This is a URL not a link?
```suggestion
* `bot_management_portal_url` - The management portal url.
```

File: azurerm/internal/services/bot/bot_healthbot_resource.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(healthbot.F0),
+					string(healthbot.S1),
+				}, false),
+			},
+
+			"bot_management_portal_link": {
```
Comment (katbyte) [inline azurerm/internal/services/bot/bot_healthbot_resource.go:63]:
```suggestion
			"bot_management_portal_url": {
```

**Review by katbyte — APPROVED** [review]
Thanks @haiyuazhang - aside from one comment about a property name this looks good - once that is changed this should be good to merge!

**Review by katbyte — APPROVED** [review]
Thanks @haiyuazhang - aside from one comment about a property name this looks good - once that is changed this should be good to merge!

**Pull request merged**